File: server/Iam.DomainService/Shared/Entities/Organization.cs
using Blocks.Genesis;
using MongoDB.Bson.Serialization.Attributes;

namespace Iam.DomainService.Shared.Entities
{
    [BsonIgnoreExtraElements]
    public class Organization : BaseEntity
    {
        public required string Name { get; set; }
        public string? Description { get; set; }
        public string? ParentOrganizationId { get; set; }
        public string? ShortCode { get; set; }
        public bool IsDisabled { get; set; } = false;
        public List<string> DefaultRoleForMembers { get; set; } = new List<string>();
        public List<string> DefaultPermissionsForMembers { get; set; } = new List<string>();

        // Contact
        public string? Email { get; set; }
        public string? PhoneNumber { get; set; }
        public string? WebsiteUrl { get; set; }
        public List<Address> Addresses { get; set; } = new List<Address>();

        // Branding
        public Theme? Theme { get; set; }
        public string? LogoUrl { get; set; }
        public string? LogoId { get; set; }
        public string? Industry { get; set; }
        public string TimeZone { get; set; } = "UTC";
        public string? Currency { get; set; }
        public string DateFormat { get; set; } = "yyyy-MM-dd";
        public string TimeFormat { get; set; } = "HH:mm";
        public string Locale { get; set; } = "en-US";

        // Add custom attributes to the organization
        public Dictionary<string, object> Attributes { get; set; } = new Dictionary<string, object>();
    }

    [BsonIgnoreExtraElements]
    public class Theme
    {
        public string? Name { get; set; }
        public string? PrimaryColor { get; set; }
        public string? SecondaryColor { get; set; }
        public string? TertiaryColor { get; set; }
        public Dictionary<string, object> Attributes { get; set; } = new Dictionary<string, object>();
    }

    [BsonIgnoreExtraElements]
    public class Address
    {
        public string Name { get; set; } = string.Empty;
        public string? AddressLine1 { get; set; }
        public string? AddressLine2 { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? PostalCode { get; set; }
        public string? Country { get; set; }
        public bool IsPrimary { get; set; }
    }
}

File: client/app/idp/iam/models/organization.ts
export interface IOrganization {
  itemId: string;
  name: string;
  description: string | null;
  parentOrganizationId: string | null;
  shortCode: string | null;
  isDisabled: boolean;
  defaultRoleForMembers: string[];
  defaultPermissionsForMembers: string[];
  email: string | null;
  phoneNumber: string | null;
  websiteUrl: string | null;
  addresses: unknown[];
  theme: string | null;
  logoUrl: string | null;
  logoId: string | null;
  industry: string | null;
  timeZone: string;
  currency: string | null;
  dateFormat: string;
  timeFormat: string;
  locale: string;
  attributes: Record<string, unknown>;
  createdDate: string;
  lastUpdatedDate: string;
  createdBy: string;
  language: string | null;
  lastUpdatedBy: string;
  organizationId: string;
  organizationIds?: string[];
  tags: string[];
}

export interface IOrganizationFilter {
  projectKey?: string;
  page: number;
  pageSize: number;
  search?: string;
  sort?: {
    property: string;
    isDescending: boolean;
  };
}

export interface IGetOrganizationsParams {
  page: number;
  pageSize: number;
  search?: string;
  isDisabled?: boolean;
  parentOrganizationId?: string;
  sort?: {
    property: string;
    isDescending: boolean;
  };
}

export interface IGetOrganizationsResponse {
  organizations: IOrganization[];
  errors: unknown;
  isSuccess: boolean;
  totalCount: number;
}

export interface IGetOrganizationByIdParams {
  projectKey: string;
  itemId: string;
}

export interface IGetOrganizationByIdResponse {
  organization: IOrganization;
  errors: unknown;
  isSuccess: boolean;
}

export interface ICreateOrUpdateOrganizationPayload {
  name: string;
  createdFrom?: number;
}

export interface IUpdateOrganizationPayload {
  itemId: string;
  name: string;
  isEnable: boolean;
}

export interface ICreateOrUpdateOrganizationResponse {
  errors: unknown;
  isSuccess: boolean;
}

export interface IMyOrganization {
  itemId: string;
  name: string;
  createdDate: string;
}

export interface IGetMyOrganizationsResponse {
  organizations: IMyOrganization[];
  errors: unknown;
  isSuccess: boolean;
}
